pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -124,90 +122,70 @@ func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
 
-	// if no events are trigger continue the execution
-	if !eventsTriggered {
-		return s.handle(&message{Msg: msg, ThreadID: thid, NextStateName: next.Name()})
-	}
-
-	return nil
+	// if no action event is triggered, continue the execution
+	return s.handle(&message{Msg: msg, ThreadID: thid, NextStateName: next.Name()})
 }
 
-// RegisterEvent on DID Exchange protocol messages. The events are triggered for incoming ConnectionRequest,
-// ConnectionResponse or ConnectionAck message types. The consumer need to invoke the callback to resume processing.
-// Only one channel can be registered for the action events. If called multiple times, the events will be sent to the
-// last channel. This works in conjunction with RegisterAutoExecute(). If this function is called after
-// RegisterAutoExecute(), the service will not execute the processing automatically and it'll wait for the callback. If
-// both are not set, then service will not handle the messages and throw an error.
-func (s *Service) RegisterEvent(ch chan<- event.DIDCommEvent) error {
+// RegisterActionEvent on DID Exchange protocol messages. The events are triggered for incoming message types based on
+// canTriggerActionEvents() function. The consumer need to invoke the callback to resume processing.
+// Only one channel can be registered for the action events. The function will throw error if a channel is already
+// registered. The AutoExecuteActionEvent() function can be used to automatically trigger callback function for the
+// event.
+func (s *Service) RegisterActionEvent(ch chan<- dispatcher.DIDCommEvent) error {
 	s.lock.Lock()
-	s.actionEvent = ch
-	s.execute = true
-	s.lock.Unlock()
+	defer s.lock.Unlock()
 
-	return nil
-}
-
-// UnregisterEvent on DID Exchange protocol messages. Refer RegisterEvent().
-func (s *Service) UnregisterEvent() error {
-	return s.disableExecute()
-}
-
-// RegisterMsg on DID Exchange protocol messages. The events are triggered for incoming ConnectionRequest,
-// ConnectionResponse or ConnectionAck message types. The Callback is set to nil in the actionEvent message and service
-// won't be expecting any response from the consumers.
-func (s *Service) RegisterMsg(ch chan<- dispatcher.DIDCommMsg) error {
-	s.statusEventLock.Lock()
-	s.statusEvents = append(s.statusEvents, ch)
-	s.statusEventLock.Unlock()
+	if s.actionEvent != nil {
+		return errors.New("channel is already registered for the action event")
+	}
 
+	s.actionEvent = ch
 	return nil
 }
 
-// UnregisterMsg on DID Exchange protocol messages. Refer RegisterMsg().
-func (s *Service) UnregisterMsg(ch chan<- dispatcher.DIDCommMsg) error {
-	s.statusEventLock.Lock()
-	for i := 0; i < len(s.statusEvents); i++ {
-		if s.statusEvents[i] == ch {
-			s.statusEvents = append(s.statusEvents[:i], s.statusEvents[i+1:]...)
-			i--
-		}
+// UnregisterActionEvent on DID Exchange protocol messages. Refer RegisterActionEvent().
+func (s *Service) UnregisterActionEvent(ch chan<- dispatcher.DIDCommEvent) error {
```

Comment:
Why mismatch the channel type and the method name?

Reply:
As discussed, [updated](https://github.com/hyperledger/aries-framework-go/compare/4d12dff2d14ec87906c8c69146befde29694ebfe..3526a9756b25f098cf4801865b8a7d1f506eb69b#diff-f761c91e753eba1c42a68076df06ce12R143) to match the format RegisterXYZEvent to DIDCommXYZ.